src/goto-programs/goto_convert.cpp

```diff
@@ -602,14 +573,7 @@ void goto_convertt::convert_expression(
   goto_programt &dest,
   const irep_idt &mode)
 {
-  if(code.operands().size()!=1)
-  {
-    error().source_location=code.find_source_location();
-    error() << "expression statement takes one operand" << eom;
-    throw 0;
-  }
-
-  exprt expr=code.op0();
+  exprt expr = code.expression();
```

Comment:
Just curious: why can't this be a const reference?

Reply:
To my understanding it gets modified in the alternative branch of the if statement, by `clean_expr` and the rest of the code.